testenv/aws/aws.go

```diff
@@ -0,0 +1,242 @@
+// Copyright © 2024 Cisco Systems, Inc. and its affiliates.
+// All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package aws
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"net/url"
+	"time"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	awsconfig "github.com/aws/aws-sdk-go-v2/config"
+	"github.com/aws/aws-sdk-go-v2/service/cloudformation"
+	cloudformationtypes "github.com/aws/aws-sdk-go-v2/service/cloudformation/types"
+	"github.com/aws/aws-sdk-go-v2/service/ec2"
+	"github.com/aws/aws-sdk-go-v2/service/s3"
+
+	"github.com/openclarity/vmclarity/testenv/aws/asset"
+	"github.com/openclarity/vmclarity/testenv/types"
+	"github.com/openclarity/vmclarity/testenv/utils"
+	"github.com/openclarity/vmclarity/testenv/utils/docker"
+)
+
+type ContextKeyType string
+
+const (
+	AWSClientContextKey         ContextKeyType = "AWSClient"
+	DefaultStackCreationTimeout                = 30 * time.Minute
+	DefaultSSHPortReadyTimeout                 = 2 * time.Minute
+)
+
+var _ types.Environment = &AWSEnv{}
+
+// AWS Environment.
+type AWSEnv struct {
+	client              *cloudformation.Client
+	ec2Client           *ec2.Client
+	s3Client            *s3.Client
+	testAsset           *asset.Asset
+	server              *Server
+	stackName           string
+	templateURL         string
+	workDir             string
+	region              string
+	sshKeyPair          *utils.SSHKeyPair
+	sshPortForwardInput *utils.SSHForwardInput
+	meta                map[string]interface{}
```

Comment:
Is this field necessary? :)

Reply:
@chrisgacsal I think I remember you mentioned that this field was required, can you remind me where?

Reply:
in other envs, meta fields are used for logging, nothing more